File: frontend/src/types/media.ts
export interface MediaFile {
  id: number;
  file_path: string;
  file_name: string;
  file_extension: string;
  media_type: 'image' | 'video';
  file_size: number;
  width: number | null;
  height: number | null;
  thumbnail_url: string | null;
  checkpoint_name: string | null;
  positive_prompt: string | null;
  negative_prompt: string | null;
  sampler_name: string | null;
  scheduler: string | null;
  cfg_scale: number | null;
  steps: number | null;
  seed: number | null;
  lora_names: string[] | null;
  metadata_prompt: Record<string, unknown> | null;
  metadata_workflow: Record<string, unknown> | null;
  file_created_at: string | null;
  file_modified_at: string | null;
  created_at: string;
}

export interface PaginationMeta {
  page: number;
  per_page: number;
  total: number;
  total_pages: number;
}

export interface MediaListResponse {
  items: MediaFile[];
  pagination: PaginationMeta;
}

export interface FolderNode {
  name: string;
  path: string;
  children: FolderNode[];
  file_count: number;
}

export interface MediaQuery {
  page?: number;
  per_page?: number;
  media_type?: 'image' | 'video';
  folder?: string;
  search?: string;
  checkpoint?: string;
  sampler?: string;
  sort_by?: 'file_created_at' | 'file_name' | 'file_size';
  sort_order?: 'asc' | 'desc';
}

File: backend/app/schemas/media.py
import json
from datetime import datetime
from typing import Any

from pydantic import BaseModel, field_validator


class MediaFileResponse(BaseModel):
    id: int
    file_path: str
    file_name: str
    file_extension: str
    media_type: str
    file_size: int
    width: int | None = None
    height: int | None = None
    thumbnail_url: str | None = None

    checkpoint_name: str | None = None
    positive_prompt: str | None = None
    negative_prompt: str | None = None
    sampler_name: str | None = None
    scheduler: str | None = None
    cfg_scale: float | None = None
    steps: int | None = None
    seed: int | None = None
    lora_names: list[str] | None = None

    metadata_prompt: dict[str, Any] | None = None
    metadata_workflow: dict[str, Any] | None = None

    file_created_at: datetime | None = None
    file_modified_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}

    @field_validator("metadata_prompt", "metadata_workflow", mode="before")
    @classmethod
    def _parse_json_string(cls, v: object) -> dict[str, Any] | None:
        if isinstance(v, str):
            try:
                parsed = json.loads(v)
                if isinstance(parsed, dict):
                    return parsed
            except (json.JSONDecodeError, TypeError):
                pass
            return None
        return v


class PaginationMeta(BaseModel):
    page: int
    per_page: int
    total: int
    total_pages: int


class MediaListResponse(BaseModel):
    items: list[MediaFileResponse]
    pagination: PaginationMeta


class FolderNode(BaseModel):
    name: str
    path: str
    children: list["FolderNode"] = []
    file_count: int = 0
